pkg/controller/workload/workload_processor.go

```diff
@@ -466,7 +496,55 @@ func (p *Processor) storeServiceFrontendData(serviceId uint32, service *workload
 	return nil
 }
 
-func (p *Processor) storeServiceData(serviceName string, waypoint *workloadapi.GatewayAddress, ports []*workloadapi.Port) error {
+func (p *Processor) updateEndpoint(serviceId uint32, toLLb bool) error {
+	var (
+		err  error
+		prio uint32
+	)
+	eksDelete := []bpf.EndpointKey{}
+	backendUids := []uint32{}
+	for _, ep := range p.EndpointCache.List(serviceId) {
+		ek := bpf.EndpointKey{
+			ServiceId:    ep.ServiceId,
+			Prio:         ep.Prio,
+			BackendIndex: ep.BackendIndex,
+		}
+		ev := bpf.EndpointValue{}
+		if err := p.bpf.EndpointLookup(&ek, &ev); err != nil { // get backend Uid
+			return err
+		}
+		backendUids = append(backendUids, ev.BackendUid)
+		eksDelete = append(eksDelete, ek)
+	}
+	if err := p.deleteEndpointRecords(eksDelete); err != nil {
```

Comment:
I am concerned about this may cause outage

Reply:
deleteEndpointRecords maintains newest bpf map state in it.